ghostfolio/ghostfolio
# Feature/expose portfolio snapshot calculation timeout as environment variable

Expose `PROCESSOR_PORTFOLIO_SNAPSHOT_COMPUTATION_TIMEOUT` env var to allow people to configure the `lockDuration` option for the Portfolio Snapshot queue. `@nodejs/bull`'s default is 30s, so if a job holds the JS event loop longer than that it gets marked as stalled ([reference](https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#advanced-settings)). The end-user experience right now for when that happens is that the ghostfolio container crashes.

Test Plan:
1. Before this diff, my container always crashes since my portfolio snapshot computation takes about 2 minutes
2. Apply this diff, set `PROCESSOR_PORTFOLIO_SNAPSHOT_COMPUTATION_TIMEOUT` to 5 minutes (`300000`)
3. Open Accounts page to trigger portfolio snapshot computation
4. Check it no longer crashes
